### PR TITLE
Enable viper for configuration management

### PR DESCRIPTION
**Pull request by supreeth7**
### What type of PR is this?
Feature

### What this PR does / Why we need it?
- This PR leverages [viper](https://github.com/spf13/viper) to enable effective configuration management for Backplane CLI.
- Users no longer need to switch between different config files based on the OCM environment, the `Backplane URL` will be dynamically populated based on the OCM environment the user is logged into. UX win!
- The configuration filepath, proxy, Backplane URL set by the users via the `BACKPLANE_CONFIG`, `HTTPS_PROXY` and `BACKPLANE_URL` environment variables are respected and given preference over the values defined in the configuration file.

### Which Jira/Github issue(s) does this PR fix?

Resolves: [OSD-15922](https://issues.redhat.com/browse/OSD-15922)

### Special notes for your reviewer
In order to test this PR, please create the `config.env` file in `$HOME/.config/backplane` folder. The contents of the file should be as follows:
```
➜ cat .config/backplane/config.env
HTTPS_PROXY="${PROXY}"

# OCM URLs based on environment
OCM_PROD="${OCM_URL}"
OCM_STG="${OCM_URL}"
OCM_INT="${OCM_URL}"

# Backplane URLs
BACKPLANE_URL_PROD="${BP_URL}"
BACKPLANE_URL_STG="${BP_URL}"
BACKPLANE_URL_INT="${BP_URL}"
```

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [x] Included documentation changes with PR

### CODE REVIEW

File: pkg/cli/config/config_test.go [hunk 1]
```diff
@@ -1,35 +1,23 @@
...
-			t.Errorf("expected path to be %v, got %v", "~/.backplane.stg.env.json", path)
-		}
-	})
 
-	t.Run("it returns the default configuration file path if it does not exist in the user's env", func(t *testing.T) {
-		path, err := GetBackplaneConfigFile()
-		if err != nil {
-			t.Error(err)
-		}
+		fmt.Println(config.ProxyURL)
```
Comment (samanthajayasinghe) [inline pkg/cli/config/config_test.go:17]: hope this is a test print

Comment (supreeth7) [reply pkg/cli/config/config_test.go:17]: Ah yes, the real way to test things :wink:

**Comment by supreeth7** [comment]
Will have to change a few things as we cannot have `.env` files, they can be easily sourced, and sensitive data can be revealed to the terminal. Additionally, an erroneous rebase, I'll be closing this PR and opening a new one.

**Pull request closed without merging**